ui/tests/e2e/lifecycle_enviroments.spec.js

```diff
@@ -0,0 +1,63 @@
+import { expect, test as base } from '@playwright/test';
+import { EnvironmentPage } from './fixtures/environmentsPage';
+import { ENV } from './env';
+
+export const test = base.extend({
+  environmentPage: async ({ page }, use) => {
+    const envPage = new EnvironmentPage(page);
+    await envPage.navigate();
+    await use(envPage);
+  },
+});
+
+test.describe('Lifecycle Environments Tests', () => {
+  const environmentName = 'Sample-playwright-test';
+
+  test('Create a Environment', async ({ environmentPage }) => {
+    await environmentPage.navigate();
+    await environmentPage.createEnvironmentProfile(environmentName);
+    await expect(environmentPage.page).toHaveURL(
+      `${ENV.MESHERY_SERVER_URL}/management/environments`,
+    );
+  });
+
+  test('Assign Connections', async ({ environmentPage }) => {
+    await environmentPage.navigate();
+    await environmentPage.assignConnections(environmentName);
+    await expect(environmentPage.page).toHaveURL(
+      `${ENV.MESHERY_SERVER_URL}/management/environments`,
+    );
+  });
+
+  test('Move Connections From Assigned To Available', async ({ environmentPage }) => {
+    await environmentPage.navigate();
+    await environmentPage.moveAssignedConnectionsToAvailable(environmentName);
+    await expect(environmentPage.page).toHaveURL(
+      `${ENV.MESHERY_SERVER_URL}/management/environments`,
+    );
+  });
+
+  test('Assign Environment To Connection', async ({ environmentPage }) => {
+    await environmentPage.goToConnections();
+    await environmentPage.addEnvironmentsToConnection(environmentName);
+    await expect(environmentPage.page).toHaveURL(
+      `${ENV.MESHERY_SERVER_URL}/management/connections`,
+    );
+  });
+
+  test('Edit Environment Card', async ({ environmentPage }) => {
+    await environmentPage.navigate();
+    await environmentPage.editEnvironmentCard(environmentName);
+    await expect(environmentPage.page).toHaveURL(
+      `${ENV.MESHERY_SERVER_URL}/management/environments`,
+    );
+  });
+
+  test('Delete Enviroment profile', async ({ environmentPage }) => {
+    await environmentPage.navigate();
+    await environmentPage.deleteEnvironmentCard(environmentName);
+    await expect(environmentPage.page).toHaveURL(
+      `${ENV.MESHERY_SERVER_URL}/management/environments`,
```

Comment:
Why assert on page url?

Reply:
URL assertions ensure the user is on the correct page after an action, validating navigation success